From: Tobias Heider <me@tobhe.de>
Date: Thu, 30 Mar 2023 14:50:25 +0200
Subject: first-boot depends on the file system being mounted.

---
 systemd/first-boot.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/systemd/first-boot.service b/systemd/first-boot.service
index 9ae9080..9547cfc 100644
--- a/systemd/first-boot.service
+++ b/systemd/first-boot.service
@@ -4,6 +4,7 @@
 Description=Sets up system on first boot
 Before=sshd.service first-boot-complete.target
 Wants=first-boot-complete.target
+After=local-fs.target
 ConditionPathIsReadWrite=/etc
 ConditionFirstBoot=yes
 
