From: Tobias Heider <me@tobhe.de>
Date: Sun, 12 Feb 2023 01:23:08 +0100
Subject: Only include the newest device trees in bootloader

Last-Updated: 2022-10-31
---
 update-m1n1 | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/update-m1n1 b/update-m1n1
index f0a4af2..d9f5480 100755
--- a/update-m1n1
+++ b/update-m1n1
@@ -42,7 +42,8 @@ else
 	fi
 fi
 
-DTBS=/lib/firmware/*-asahi/device-tree/apple/*.dtb
+KPATHS=(/lib/firmware/*/device-tree)
+DTBS=${KPATHS[-1]}/apple/*.dtb
 
 if [ ! -e "${uboot_src}/u-boot-nodtb.bin" ]; then
 	echo "${uboot_src}/u-boot-nodtb.bin not found" 1>&2
