From: Tobias Heider <me@tobhe.de>
Date: Sun, 12 Feb 2023 01:23:08 +0100
Subject: Remove Arch Linux specific commands

Last-Update: 2022-09-07
---
 first-boot | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/first-boot b/first-boot
index e5be574..d885a4d 100755
--- a/first-boot
+++ b/first-boot
@@ -39,9 +39,3 @@ EOF
 fi
 
 update-grub
-
-echo
-echo "Initializing Pacman keyring..."
-pacman-key --init
-pacman-key --populate archlinuxarm asahilinux
-echo
