From: Tobias Heider <me@tobhe.de>
Date: Sun, 12 Feb 2023 01:23:08 +0100
Subject: Use bash, dash does not support process substitution

Last-Update: 2022-09-10
---
 update-vendor-firmware | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update-vendor-firmware b/update-vendor-firmware
index ef63628..1659753 100755
--- a/update-vendor-firmware
+++ b/update-vendor-firmware
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # SPDX-License-Identifier: MIT
 
 set -e
