From: Tobias Heider <me@tobhe.de>
Date: Sun, 12 Feb 2023 01:23:08 +0100
Subject: Update paths for ubuntu.

===================================================================
---
 update-m1n1 | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/update-m1n1 b/update-m1n1
index 841b0af..d5d8668 100755
--- a/update-m1n1
+++ b/update-m1n1
@@ -11,11 +11,10 @@ else
     . /usr/share/asahi-scripts/functions.sh
 fi
 
-: ${SOURCE:="/usr/lib/asahi-boot/"}
-: ${M1N1:="$SOURCE/m1n1.bin"}
-: ${U_BOOT:="$SOURCE/u-boot-nodtb.bin"}
+: ${M1N1:="/usr/share/m1n1/m1n1.bin"}
+: ${U_BOOT:="/usr/share/u-boot-asahi/u-boot-nodtb.bin"}
 : ${TARGET:="$1"}
-: ${DTBS:=/lib/modules/*-ARCH/dtbs/*.dtb}
+: ${DTBS:=/lib/firmware/*-asahi/device-tree/apple/*.dtb}
 : ${CONFIG:=/etc/m1n1.conf}
 
 umount=false
