Description: Use bash, dash does not support process substitution
Author: Tobias Heider <me@tobhe.de>
Last-Update: 2022-09-10

Index: asahi-scripts/update-vendor-firmware
===================================================================
--- asahi-scripts.orig/update-vendor-firmware
+++ asahi-scripts/update-vendor-firmware
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # SPDX-License-Identifier: MIT
 
 set -e
