From: Tobias Heider <me@tobhe.de>
Date: Sun, 12 Feb 2023 01:23:08 +0100
Subject: ubuntu-fix-first-boot

===================================================================
---
 first-boot | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/first-boot b/first-boot
index d885a4d..2cfc067 100755
--- a/first-boot
+++ b/first-boot
@@ -5,6 +5,7 @@ set -e
 
 root_dev=$(findmnt -n -o SOURCE /)
 efi_dev=$(findmnt -n -o SOURCE /boot/efi)
+root_uuid_old=$(findmnt -n -o UUID /)
 
 if [ -e "$root_dev" ]; then
     echo "Randomizing root filesystem UUID..."
@@ -38,4 +39,14 @@ EOF
     echo
 fi
 
+# Ubuntu doesn't ship systemd-firstboot(1), improvise
+systemd-machine-id-setup
+
+# XXX: grub-probe will look for the old UUID, this will keep it happy.
+#      sleep 1 is needed to make sure the new dev exists.
+if [ ! -z "$root_uuid" ]; then
+    sleep 1
+    ln -s "/dev/disk/by-uuid/$(echo "$root_uuid" | cut -d "=" -f2)" "/dev/disk/by-uuid/$root_uuid_old"
+fi
+
 update-grub
